### DISCUSSION TITLE
ShareInto: Add timestamp placeholder to ShareIntoPrefix

### DISCUSSION THREAD

**Discussion by bruvellu** [gsantner/markor#2094] [Ideas] [answered]
Hello,

I've been using Markor to edit my Zim notes on my mobile, and it's been a delight.

The Share Into feature to QuickNotes is really handy for capturing external content into my notebook. But for me, it's important to know when I saved a particular content. Currently, I manually add a timestamp by opening the quick notes file and using the date/time action. But it would be amazing if the timestamp was automatically saved with the content.

One idea I was thinking is that it could be added to the [ShareIntoPrefix](https://github.com/gsantner/markor/blob/d8ffc01fddb6c833e9e00a8d52151d9d670684bf/app/src/main/java/net/gsantner/markor/model/AppSettings.java#L899) as a placeholder. Something like this:

```
\\n----\\n{{timestamp}}\\n{{text}}
```

Is this something that would be useful for others (or even feasible)?

Or perhaps there's already a solution that I overlooked?

Thanks!
Bruno

**Comment by harshad1** [comment, marked as answer]
Simpledateformat in backticks will be interpolated

"\\n----\\n\`yyyy-MM-dd\`\\n{{text}}"

**Reply by bruvellu** [reply]
Oh, that's perfect! Thanks, @harshad1. After some testing, I noticed the backticks aren't needed, they are just rendered as backticks.

To get an ISO timestamp with offset, I used:
```
\n----\n\nyyyy-MM-dd'T'HH:mm:ssXXX\n\n{{text}}
```
Which renders to:
```
2023-08-24T21:14:20+02:00
```

**Comment by harshad1** [comment]
FYI  https://docs.oracle.com/javase/8/docs/api/java/text/SimpleDateFormat.html
